internal/policy/root.go

```diff
@@ -27,6 +27,49 @@ func InitializeRootMetadata(key *tuf.Key) *tuf.RootMetadata {
 	return rootMetadata
 }
 
+func AddRootKey(rootMetadata *tuf.RootMetadata, rootKey *tuf.Key) *tuf.RootMetadata {
+	rootMetadata.Keys[rootKey.KeyID] = rootKey
+	if _, ok := rootMetadata.Roles[RootRoleName]; !ok {
+		return InitializeRootMetadata(rootKey)
+	}
+
+	rootMetadata.AddKey(rootKey)
```

Comment:
isn't this fundamentally the same as line 31? I suggest removing line 31 in that case.